src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c

```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK1,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
+    { 18, 64, 0, { 0, 0, 0 } },
+};
+
+/* Animations struct */
+static struct_80B8E1A8 D_809CDC7C[4] = {
+    { &D_06000FDC, 1.0f, 0, 0.0f }, /* Looking through telescope */
+    { &D_06005B58, 1.0f, 0, 10.0f }, /* Breathing? Unused? */
+    { &D_06000AB0, 1.0f, 0, 0.0f }, /* Talking */
+    { &D_0600066C, 1.0f, 2, -5.0f }, /* Scratching chin? */
+};
+
+
+void func_809CCDE0(EnBji01* this, GlobalContext* globalCtx) {
+
+    Player* player = PLAYER;
+
+    Vec3f sp58;
+    s32 pad[2];
+
+    Math_Vec3f_Copy(&sp58, &player->actor.world.pos);
+    sp58.y = player->bodyPartsPos[7].y + 3.0f;
+    func_8013E950(&this->actor.world.pos, &this->actor.focus.pos, this->actor.shape.rot.y, &player->actor.world.pos,
+                  &sp58, &this->headZRotAdj, &this->headXRotAdj, &this->torsoZRotAdj, &this->torsoXRotAdj, 0x1554, 0x1FFE, 0xE38, 0x1C70);
+}
+
+void func_809CCE98(EnBji01* this, GlobalContext* globalCtx) {
+
+    func_8013E1C8(&this->skelAnime, D_809CDC7C, 0, &this->animationIndex);
+    this->actor.textId = 0;
+    this->actionFunc = func_809CCEE8;
+}
+
+void func_809CCEE8(EnBji01* this, GlobalContext* globalCtx) {
+
+    Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y, 0x444);
+    if (this->actor.params == 0) {
+        if ((this->actor.xzDistToPlayer <= 60.0f) && (this->actor.yDistToPlayer <= 10.0f)) {
+            this->actor.flags |= 0x10000;
+        } else {
+            this->actor.flags &= 0xFFFEFFFF;
+        }
+    }
+    if (func_800B84D0(&this->actor, globalCtx) != 0) {
+        globalCtx->msgCtx.unk11F22 = 0;
+        globalCtx->msgCtx.unk11F10 = 0;
+        func_809CD028(this, globalCtx);
+    } else {
+        if (this->moonsTear != NULL) {
+            if (this->moonsTear->actor.colChkInfo.health == 1) {
+                func_809CD6C0(this, globalCtx);
+                return;
+            }
+        } else {
+            this->moonsTear =
+                (ObjMoonStone*)func_ActorCategoryIterateById(globalCtx, NULL, ACTORCAT_PROP, ACTOR_OBJ_MOON_STONE);
+        }
+        func_800B8500(&this->actor, globalCtx, 60.0f, 10.0f, 0);
+    }
+}
+
+void func_809CD028(EnBji01* this, GlobalContext* globalCtx) {
+
+    s32 tempDay;
+    f32 tempTimeBeforeMoonCrash;
+
+    switch (this->actor.params) {
+        case 0: /* Have not talked to Shikashi as of entering observatory last */
+        case 1: /* Have finished a conversation with Shikashi */
+            switch (gSaveContext.playerForm) {
+                case PLAYER_FORM_DEKU:
+                    if ((gSaveContext.weekEventReg[17] & 0x10) != 0) {
+                        if ((gSaveContext.weekEventReg[74] & 0x80) != 0) {
+                            this->textId = 0x5F4;
```

Comment:
There's no need to leave the extra `!= 0` for those bitwise checks.
```suggestion
                    if (gSaveContext.weekEventReg[17] & 0x10) {
                        if (gSaveContext.weekEventReg[74] & 0x80) {
                            this->textId = 0x5F4;
```